Add new docs structure (#26)
exercism/v3: docs/maintainers/README.md

# Maintainers - How to get started

> Firstly, thank you for your willingness to help with Exercism. We believe that v3 will be a huge step forward in our aim to offer a comprehesive free learning website to anyone wanting to develop their programming skills, and we're really excited to see it play out.

Reviewer: ```suggestion
Firstly, thank you for your willingness to help with Exercism. We believe that v3 will be a huge step forward in our aim to offer a comprehensive, free learning website to anyone wanting to develop their programming skills, and we're really excited to see it play out.
```

**Before doing anything else, please create an issue in this repository asking for your track to be bootstrapped. You can then spend some time learning about the v3 process while we set things up for you.**

## Concept Exercises

> The key things to understand about v3 is that we are making a strong distinction between Concept Exercises and Practice Exercises. All existing exercises will become Practice Exericses - exercises where students can practice the ideas they are learning, and discuss them with a mentor. The new Concept Exercises will be written from scratch, each with the aim of teaching one or more programming concepts that are necessary to become fluent in a given language.

Reviewer: ```suggestion
The key thing to understand about v3 is that we are making a strong distinction between Concept Exercises and Practice Exercises. All existing exercises will become Practice Exericses - exercises where students can practice the ideas they are learning, and discuss them with a mentor. The new Concept Exercises will be written from scratch, each with the aim of teaching one or more programming concepts that are necessary to become fluent in a given language.
```

Maybe instead of `can practice the ideas they are learning` do `can practice the concepts they are learning`?

### Read more

- [More about Concepts and Concept Exercises](../concept-exercises.md)
- [How to determine Concepts for your track](./determining-concepts.md)
- [How to write Concept Exercise GitHub issues](./writing-a-concept-exercise-github-issue.md)
- [How the repository is structured](./repository-structure.md)

## Migrating your config.json files

For v3, the existing `config.json` files will need to be updated. You can use the [C# config.json](../../languages/csharp/config.json) for reference.

1. Add a version property:

```json
"version": 3
```

2. Add online editor settings:

TODO: Write up something about https://github.com/exercism/ruby/blob/master/config.json#L7

3. Convert the `"exercises"` array to an object with two properties:

```json
"exercises": {
  "concept": [],
  "practice": []
}
```

4. Added concepts for all Practice Exercises

The existing exercises are temporarily removed from the `config.json` file. They will return as practice exercises once the concept exercises have been added.

> TODO: Discuss more about what needs to be done

Reviewer: @iHiD Do we need to write this bit already? I know what I want the new format to look like, so I could do that.

## Tooling

As well as building out the Concept Exercises, your track will also need a Test Runner and a Representer.

> The Test Runner is a piece of software that runs the tests. This enables us to have in-browser coding. Test Runners are surprisingly tricky to get right, with lots of weird edge cases and oddities, so we recommend starting on them quite early. You can find more in the [automated testing repository](https://github.com/exercism/automated-tests).

Reviewer: ```suggestion
The Test Runner is a piece of software that runs the tests. This enables us to have in-browser coding. Test Runners are surprisingly tricky to get right, with lots of weird edge cases and oddities, so we recommend starting on them quite early. You can learn more about Test Runners in the [automated testing repository](https://github.com/exercism/automated-tests).
```

Representers are pieces of software that take a solution and provide a normalised representation of it. We can then attach feedback to those representations, and automatically provide the same feedback for future solutions that normalise to the same representation. You may also like to create an Analyzer, which automatically provides feedback based on heursitics provided by AST matching. You can learn more about Representers and Analysers in our [automated analysis repository](https://github.com/exercism/automated-analysis).

## Getting help

> Each track has a #maintaining-x channel on [Slack](https://exercism-team.slack.com). We recommend using that to chat amongst yourselves.

Reviewer: I'm wondering a little bit if we should perhaps also encourage people to post comments on the GH issues, as then the discussion becomes public and is automatically archived.
